packages/framework/src/hooks/useThemeContext.tsx

```diff
@@ -0,0 +1,56 @@
+import { useAtom, useAtomValue } from "jotai";
+import { clone } from "lodash-es";
+import React, { createContext, useCallback, useContext } from "react";
+import { type EnsembleThemeModel } from "../shared";
+import { appAtom, selectedThemeNameAtom, themeAtom } from "../state";
+
+export type CustomTheme = { [key: string]: unknown } & {
+  theme?: EnsembleThemeModel;
+  themeName?: string;
+  setTheme?: (name: string) => void;
+};
+export const CustomThemeContext = createContext<CustomTheme>({});
+
+type CustomThemeProps = {
+  value?: CustomTheme;
+} & React.PropsWithChildren<CustomTheme>;
+export const CustomThemeProvider: React.FC<CustomThemeProps> = ({
+  children,
+  value,
+}) => {
+  return (
+    <CustomThemeContext.Provider value={{ ...value }}>
+      {children}
+    </CustomThemeContext.Provider>
+  );
+};
+
+export const useThemeScope = (): {
```

Comment:
Can we consolidate this hook into a single one?  One provider, one hook to access provider values.

Reply:
No, we can't consolidate this into single one, because for provider, we need two hook.

Reply:
You only need one custom hook, the other is provided by react out of the box.